staging/src/k8s.io/cli-runtime/pkg/resource/query_param_verifier_v3.go

```diff
@@ -124,9 +125,19 @@ func supportsQueryParamV3(doc *spec3.OpenAPI, gvk schema.GroupVersionKind, query
 		// Now look for the query parameter among the parameters
 		// for the PATCH operation.
 		for _, param := range op.OperationProps.Parameters {
-			if param.ParameterProps.Name == string(queryParam) {
+			if param.ParameterProps.Name == string(queryParam) && param.In == "query" {
 				return nil
 			}
+
+			// lookup global parameters
+			if ref := param.Refable.Ref.Ref.String(); strings.HasPrefix(ref, "#/parameters/") && doc.Components != nil {
+				k := strings.TrimPrefix(ref, "#/parameters/")
+				if globalParam, ok := doc.Components.Parameters[k]; ok {
+					if globalParam.In == "query" && globalParam.Name == string(queryParam) {
```

Comment:
I don't have a good sense for which of these can be nil in weird schemas... do we need to guard for nil in `doc.Components` or `globalParam` values?

Reply:
`doc.Components` is guarded already. Have also added the same for `globalParam`. Don't think a null parameter is possible. But safe code never hurts.